webapp/src/utils/adapters/index.ts

```diff
@@ -27,3 +31,51 @@ export interface IMessageAdapter {
 export interface ITranslationAdapter {
   getTranslations(): Promise<TranslationMap>;
 }
+
+/** convert { [id]: { file, texts } } to { [id]: text } */
+export function dehydrateMessageMap(
+  translations: MessageMap,
+): Record<string, string> {
+  const output: Record<string, string> = {};
+  Object.entries(translations).forEach(([id, { text }]) => {
+    output[id] = text;
+  });
+  return output;
+}
+
+type TranslationFilenameMap = Record<
+  string, // lang
+  Record<
+    string, // filename
+    Record<
+      string, // msg Id
+      string // text
+    >
+  >
+>;
+
+/** convert TranslateMap to { [lang]: { [filename]: {[msgId]: text} } */
+export function groupByFilename(
+  translationMap: TranslationMap,
+): TranslationFilenameMap {
+  const output: TranslationFilenameMap = {};
+  Object.entries(translationMap).forEach(([lang, messageMap]) => {
+    Object.entries(messageMap).forEach(([msgId, { text, sourceFile }]) => {
+      if (!output[lang]) {
+        output[lang] = {};
+      }
+      if (!output[lang][sourceFile]) {
+        output[lang][sourceFile] = {};
+      }
+      const shortId = getShortId(msgId, sourceFile);
+      output[lang][sourceFile][shortId] = text;
+    });
+  });
+  return output;
+}
+
+function getShortId(msgId: string, sourceFile: string): string {
+  const sourceFileDotCount = sourceFile.replace('/', '.').split('.').length - 1;
```

Comment:
`String.prototype.replace` only replaces the first occurrence if pattern is a string.

For example `getShortId('a.b.c.d', 'a/b/c/sv.yml')` should return just 'd' but this implementation returns 'c.d'.

Fortunately, there is another function `replaceAll` that works as expected.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll